api/nntrainer-api-common.h

```diff
@@ -37,7 +37,29 @@ typedef enum {
   ML_TRAIN_LAYER_TYPE_ADDITION,   /**< Addition Layer (Since 6.5) */
   ML_TRAIN_LAYER_TYPE_CONCAT,     /**< Concat Layer (Since 6.5) */
   ML_TRAIN_LAYER_TYPE_MULTIOUT,   /**< MultiOut Layer (Since 6.5) */
-  ML_TRAIN_LAYER_TYPE_UNKNOWN = 999 /**< Unknown Layer */
+  ML_TRAIN_LAYER_TYPE_PREPROCESS_FLIP, /**< Preprocess flip Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_PREPROCESS_TRANSLATE, /**< Preprocess translate Layer
+                                               (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_PREPROCESS_L2NORM, /**< Preprocess L2Normalization Layer
+                                            (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_EMBEDDING,         /** Embedding Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_RNN,               /** RNN Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_LSTM,              /** LSTM Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_SPLIT,             /** Splite Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_GRU,               /** GRU Layer (Since 6.5) */
+  ML_TRAIN_LAYER_TYPE_TIME_DIST, /** Time Distributed Layer (Since 6.5) */
```

Comment:
Could someone else confirm this should be exposed as well?

Reply:
i this can be kept internal as for every layer, distribute can be set as a property.
@jijoongmoon can confirm this.

Reply:
Right. It can be kept internal. It is viewed as a property.